crates/nu-command/src/commands/filters/drop/nth.rs

```diff
@@ -41,18 +47,46 @@ impl WholeStreamCommand for SubCommand {
                 example: "echo [first second third] | drop nth 0 2",
                 result: Some(vec![Value::from("second")]),
             },
+            Example {
+                description: "Drop range rows from second to fourth",
+                example: "echo [first second third fourth fifth] | drop nth (1..3)",
+                result: Some(vec![Value::from("first"), Value::from("fifth")]),
+            },
         ]
     }
 }
 
+fn extract_int_or_range(args: &CommandArgs) -> Result<Either<u64, Range>, ShellError> {
+    let value = args.req::<Value>(0)?;
+
+    let int_opt = value.as_u64().map(Either::Left).ok();
+    let range_opt = FromValue::from_value(&value).map(Either::Right).ok();
+
+    int_opt
+        .or(range_opt)
+        .ok_or_else(|| ShellError::type_error("int or range", value.spanned_type_name()))
+}
+
 fn drop(args: CommandArgs) -> Result<OutputStream, ShellError> {
-    let row_number: Tagged<u64> = args.req(0)?;
-    let and_rows: Vec<Tagged<u64>> = args.rest(1)?;
-    let input = args.input;
+    let number_or_range = extract_int_or_range(&args)?;
+    let rows = match number_or_range {
+        Either::Left(row_number) => {
+            let and_rows: Vec<Tagged<u64>> = args.rest(1)?;
+
+            let mut rows: Vec<_> = and_rows.into_iter().map(|x| x.item as usize).collect();
+            rows.push(row_number as usize);
+            rows.sort_unstable();
+            rows
+        }
+        Either::Right(row_range) => {
```

Comment:
Should we fail if extra args are supplied? Right now we are silently ignoring them.

Reply:
can you say more about this? I don't remember the original design enough to know what we might be ignoring.

Reply:
@jntrnr originally drop accepts a single index with optionally multiple indices. But now the first argument can also be a range. In which the rest arguments are ignored. So my question is how should we handle rest arguments. IMO possible cases are:
1- ignore them
2- error out if there are additional arguments are given
3- also drop the additional indices given.